daemon/emer-daemon.c

```diff
@@ -141,7 +142,8 @@ typedef struct _EmerDaemonPrivate
 {
   guint network_send_interval;
   GQueue *upload_queue;
-  gboolean uploading;
+  /* Non-NULL iff an upload is in flight */
+  GCancellable *current_upload_cancellable;
```

Comment:
While looking at this I noticed that there's a theoretical case where `finish_network_callback` could call `g_signal_emit` on a freed `EmerDaemon`, so added a patch to fix that case to the top of the PR.